asdf/asdf.py

```diff
@@ -449,6 +449,9 @@ def close(self):
             self._fd.close()
             self._fd = None
             self._closed = True
+            # as we're closing the file, also empty out the
+            # tree so that references to array data can be released
+            self._tree = AsdfObject()
```

Comment:
A custom exception sounds like a great idea. Currently and `OSError` is raised so any custom exception we raise could inherit from that exception class for backwards compatibility.
https://github.com/asdf-format/asdf/blob/b43b2c5b8935c9bf99a7ead553b32b11ccbee823/asdf/asdf.py#L559-L561
It might make sense to raise the same error type when a block is accessed from a closed file (and any other places where access to a closed file is prevented):
https://github.com/asdf-format/asdf/blob/b43b2c5b8935c9bf99a7ead553b32b11ccbee823/asdf/block.py#L1290-L1292
Perhaps a `ClosedFileAccess` exception that inherits from `OSerror`? That seems reasonably backwards compatible to include in 3.0 (and would work towards cleaning up the rather generic exceptions raised by asdf).